rootfs/etc/nginx/lua/test/configuration_test.lua

```diff
@@ -0,0 +1,66 @@
+package.path = "./rootfs/etc/nginx/lua/?.lua;./rootfs/etc/nginx/lua/test/mocks/?.lua;" .. package.path
+_G._TEST = true
+
+local _ngx = {
+  shared = {
+      configuration_data = {
+        get = function(self, rsc) return {backend = true} end,
+        set = function(self, a) end,
+      }
+  },
+  print = function(msg) return msg end,
+  log = function() end,
+  var = {},
+  req = {
+      read_body = function() end,
+      get_body_data = function() return {body_data = true} end,
+  }
+}
+_G.ngx = _ngx
+
+local configuration = require("configuration")
+
+describe("Configuration", function()
+    context("Request method is neither GET nor POST", function()
+        it("should log 'Only POST and GET requests are allowed!'", function()
+            ngx.var.request_method = "PUT"
+            local s = spy.on(ngx, "print")
+            assert.has_no.errors(function() configuration.call() end)
+            assert.spy(s).was_called_with("Only POST and GET requests are allowed!")
+        end)
+    end)
+
+    context("GET request to /configuration/backends", function()
+        before_each(function()
```

Comment:
Nit: Do we need to have `ngx.var.request_method` and `ngx.var.request_uri` in the `before_each`? It looks like they don't get changed in the two `it` blocks below.

Reply:
@hnrytrn Correct, we're not using this  values on the `it` blocks below, but they are being used inside `configuration.call()` method. An example is: https://github.com/Shopify/ingress/blob/master/rootfs/etc/nginx/lua/configuration.lua#L31

Reply:
Yeah, what I meant is that I don't think the `before_each` is needed e.g.
```
context("GET request to /configuration/backends", function()
        ngx.var.request_method = "GET"
        ngx.var.request_uri = "/configuration/backends"
```

Reply:
I like the `before_each`, it keeps these two tests idempotent